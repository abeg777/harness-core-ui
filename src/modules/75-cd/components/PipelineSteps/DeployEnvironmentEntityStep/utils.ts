/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getMultiTypeFromValue, MultiTypeInputType, SelectOption } from '@harness/uicore'
import { defaultTo, get, isEmpty, isNil, set } from 'lodash-es'
import type { EnvironmentYamlV2 } from 'services/cd-ng'
import type { DeployEnvironmentEntityConfig, DeployEnvironmentEntityFormState } from './types'

export function processInitialValues(initialValues: DeployEnvironmentEntityConfig): DeployEnvironmentEntityFormState {
  const formState = {}

  if (initialValues.environment && initialValues.environment.environmentRef) {
    const infrastructureDefinition = initialValues.environment.infrastructureDefinitions?.[0]
    return {
      environment: initialValues.environment.environmentRef,
      environmentInputs:
        getMultiTypeFromValue(initialValues.environment.environmentRef) === MultiTypeInputType.FIXED
          ? { [initialValues.environment.environmentRef]: initialValues.environment?.environmentInputs }
          : {},
      infrastructure: infrastructureDefinition?.identifier,
      // ? Can there be a better condition to identify?
      ...(!!infrastructureDefinition?.identifier && {
        infrastructureInputs: !isEmpty(infrastructureDefinition?.identifier)
          ? {
              [initialValues.environment.environmentRef]: {
                [infrastructureDefinition.identifier]: infrastructureDefinition?.inputs
              }
            }
          : {}
      })
    }
  } else if (initialValues.environments) {
    defaultTo(initialValues.environments?.values, []).map((environment: EnvironmentYamlV2, index: number) => {
      set(formState, `environments.${index}`, {
        label: environment.environmentRef,
        value: environment.environmentRef
      })
      set(formState, `environmentInputs.${environment.environmentRef}`, environment.environmentInputs)
      set(formState, `serviceOverrideInputs.${environment.environmentRef}`, environment.serviceOverrideInputs)
      if (Array.isArray(environment.infrastructureDefinitions)) {
        environment.infrastructureDefinitions.map((infrastructure, infrastructureIndex) => {
          set(formState, `infrastructures.${environment.environmentRef}.${infrastructureIndex}`, {
            label: infrastructure.identifier,
            value: infrastructure.identifier
          })
          set(
            formState,
            `infrastructureInputs.${environment.environmentRef}.${infrastructure.identifier}`,
            infrastructure.inputs
          )
        })
      }
    })
  } else if (initialValues.environmentGroup && initialValues.environmentGroup.envGroupRef) {
    set(formState, `environmentGroup`, initialValues.environmentGroup.envGroupRef)

    // This section is same as multi env. Move to function
    defaultTo(initialValues.environmentGroup.environments, []).map((environment: EnvironmentYamlV2, index: number) => {
      set(formState, `environments.${index}`, {
        label: environment.environmentRef,
        value: environment.environmentRef
      })
      set(formState, `environmentInputs.${environment.environmentRef}`, environment.environmentInputs)
      set(formState, `serviceOverrideInputs.${environment.environmentRef}`, environment.serviceOverrideInputs)
      if (Array.isArray(environment.infrastructureDefinitions)) {
        environment.infrastructureDefinitions.map((infrastructure, infrastructureIndex) => {
          set(formState, `infrastructures.${environment.environmentRef}.${infrastructureIndex}`, {
            label: infrastructure.identifier,
            value: infrastructure.identifier
          })
          set(
            formState,
            `infrastructureInputs.${environment.environmentRef}.${infrastructure.identifier}`,
            infrastructure.inputs
          )
        })
      }
    })
  }

  set(formState, `parallel`, defaultTo(initialValues.environments?.metadata?.parallel, true))

  return formState
}

export function processFormValues(data: DeployEnvironmentEntityFormState): DeployEnvironmentEntityConfig {
  if (!isNil(data.environment)) {
    return {
      environment: {
        environmentRef: data.environment,
        ...(!!data.environmentInputs?.[data.environment] && {
          environmentInputs: data.environmentInputs[data.environment]
        }),
        deployToAll: false,
        ...(!!data.infrastructure && {
          infrastructureDefinitions: [
            {
              identifier: data.infrastructure,
              inputs: get(data, `infrastructureInputs.${data.environment}.${data.infrastructure}`)
            }
          ]
        })
      }
    }
  } else if (!isNil(data.environmentGroup)) {
    return {
      environmentGroup: {
        envGroupRef: data.environmentGroup,
        deployToAll: isEmpty(data.environments),
        environments: (data.environments as unknown as SelectOption[])?.map(environment => ({
          environmentRef: environment.value as string,
          ...(!!data.environmentInputs?.[environment.value as string] && {
            environmentInputs: data.environmentInputs[environment.value as string]
          }),
          deployToAll: !data.infrastructures?.[environment.value as string],
          ...(!!data.infrastructures?.[environment.value as string] && {
            infrastructureDefinitions: data.infrastructures[environment.value as string].map(infrastructure => ({
              identifier: infrastructure.value as string,
              inputs: data.infrastructureInputs?.[environment.value as string][infrastructure.value as string]
            }))
          })
        }))
      }
    }
  } else if (!isNil(data.environments)) {
    return {
      environments: {
        values: data.environments?.map(environment => ({
          environmentRef: environment.value as string,
          ...(!!data.environmentInputs?.[environment.value as string] && {
            environmentInputs: data.environmentInputs[environment.value as string]
          }),
          deployToAll: !data.infrastructures?.[environment.value as string],
          ...(!!data.infrastructures?.[environment.value as string] && {
            infrastructureDefinitions: data.infrastructures[environment.value as string].map(infrastructure => ({
              identifier: infrastructure.value as string,
              inputs: data.infrastructureInputs?.[environment.value as string][infrastructure.value as string]
            }))
          })
        }))
      }
    }
  }
  return {}
}
