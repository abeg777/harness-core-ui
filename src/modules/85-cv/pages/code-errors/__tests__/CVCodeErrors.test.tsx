/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import * as hooks from '@common/hooks/useFeatureFlag'
import { CVCodeErrors } from '../CVCodeErrors'

const WrapperComponent = (): React.ReactElement => {
  return (
    <TestWrapper>
      <CVCodeErrors />
    </TestWrapper>
  )
}

describe('Unit tests for CVCodeErrors', () => {
  test('Verify CodeErrors page renders and matches snapshot', async () => {
    const container = render(<WrapperComponent />)
    expect(container).toMatchSnapshot()
  })

  test('Verify CodeErrors page enabled is not empty ', async () => {
    const useFeatureFlags = jest.spyOn(hooks, 'useFeatureFlag')
    useFeatureFlags.mockReturnValue(true)
    const container = render(<WrapperComponent />)
    expect(container).not.toBeNull()
  })
})
