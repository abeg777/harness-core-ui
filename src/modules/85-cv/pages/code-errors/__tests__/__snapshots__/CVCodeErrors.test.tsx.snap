// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for CVCodeErrors Verify CodeErrors page renders and matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="StyledProps--font StyledProps--main"
        style="width: 100%; height: 100%;"
      >
        <div
          class="pageHeaderDiv"
        >
          <div
            class="Layout--horizontal StyledProps--main css-1nbi0sg"
          >
            <div
              class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <div
                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
              >
                <a
                  class="Breadcrumbs--breadcrumb"
                  href="/account/undefined/settings"
                >
                  <span
                    class="bp3-icon bp3-icon-cog StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-primary7"
                    icon="cog"
                  >
                    <svg
                      data-icon="cog"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        cog
                      </desc>
                      <path
                        d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                  >
                    common.accountSettings
                  </p>
                </a>
                <span
                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                  data-icon="main-chevron-right"
                >
                  <svg
                    height="8"
                    viewBox="0 0 8 14"
                    width="8"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                      fill="#4d5969"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
          <h3
            class="pageHeaderTitle"
          >
            cv.codeErrors
          </h3>
          <p
            class="pageHeaderText"
          >
            cv.codeErrorsHeading
          </p>
        </div>
        <div
          class="StyledProps--font StyledProps--main PageSpinner--spinner StyledProps--flex StyledProps--flex-align-center-center"
        >
          <div
            class="Layout--vertical Layout--layout-spacing-medium StyledProps--main PageSpinner--content"
            style="align-items: center; width: 500px;"
          >
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
              data-icon="steps-spinner"
            >
              <svg
                height="32"
                viewBox="0 0 128 128"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M122.5 69.25H96.47a33.1 33.1 0 000-10.5h26.05a5.25 5.25 0 010 10.5z"
                    fill="#ff8e02"
                  />
                  <path
                    d="M112.04 97.83L89.47 84.8a33.1 33.1 0 005.25-9.1l22.57 13.03a5.25 5.25 0 01-5.28 9.1zM88.68 117.35L75.65 94.78a33.1 33.1 0 009.1-5.25l13.02 22.57a5.25 5.25 0 11-9.1 5.25zM58.7 122.57V96.5a33.1 33.1 0 0010.5 0v26.07a5.25 5.25 0 01-10.5 0zM30.1 112.1l13.04-22.57a33.1 33.1 0 009.1 5.25L39.2 117.35a5.25 5.25 0 11-9.1-5.25zM10.6 88.74L33.16 75.7a33.1 33.1 0 005.25 9.1L15.88 97.83a5.25 5.25 0 11-5.25-9.1z"
                    fill="#ffddb3"
                    fill-opacity="0.3"
                  />
                  <path
                    d="M5.37 58.75h26.06a33.1 33.1 0 000 10.5H5.37a5.25 5.25 0 010-10.5z"
                    fill="#ffd29a"
                    fill-opacity="0.4"
                  />
                  <path
                    d="M15.85 30.17L38.4 43.2a33.1 33.1 0 00-5.24 9.1L10.6 39.25a5.25 5.25 0 115.25-9.1z"
                    fill="#ffc680"
                    fill-opacity="0.5"
                  />
                  <path
                    d="M39.2 10.65l13.03 22.57a33.1 33.1 0 00-9.1 5.25l-13-22.57a5.25 5.25 0 119.1-5.25z"
                    fill="#ffbb67"
                    fill-opacity="0.6"
                  />
                  <path
                    d="M69.2 5.43V31.5a33.1 33.1 0 00-10.5 0V5.42a5.25 5.25 0 1110.5 0z"
                    fill="#ffb04d"
                    fill-opacity="0.7"
                  />
                  <path
                    d="M97.77 15.9L84.75 38.47a33.1 33.1 0 00-9.1-5.25l13.03-22.57a5.25 5.25 0 119.1 5.25z"
                    fill="#ffa535"
                    fill-opacity="0.8"
                  />
                  <path
                    d="M117.3 39.26L94.7 52.3a33.1 33.1 0 00-5.25-9.1l22.57-13.03a5.25 5.25 0 015.25 9.1z"
                    fill="#ff991b"
                    fill-opacity="0.9"
                  />
                  <animatetransform
                    attributeName="transform"
                    calcMode="discrete"
                    dur="1200ms"
                    repeatCount="indefinite"
                    type="rotate"
                    values="0 64 64;30 64 64;60 64 64;90 64 64;120 64 64;150 64 64;180 64 64;210 64 64;240 64 64;270 64 64;300 64 64;330 64 64"
                  />
                </g>
              </svg>
            </span>
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--font-align-center StyledProps--color StyledProps--color-grey600"
            >
              Loading, please wait...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledProps--font StyledProps--main"
      style="width: 100%; height: 100%;"
    >
      <div
        class="pageHeaderDiv"
      >
        <div
          class="Layout--horizontal StyledProps--main css-1nbi0sg"
        >
          <div
            class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <a
                class="Breadcrumbs--breadcrumb"
                href="/account/undefined/settings"
              >
                <span
                  class="bp3-icon bp3-icon-cog StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-primary7"
                  icon="cog"
                >
                  <svg
                    data-icon="cog"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      cog
                    </desc>
                    <path
                      d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                >
                  common.accountSettings
                </p>
              </a>
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                data-icon="main-chevron-right"
              >
                <svg
                  height="8"
                  viewBox="0 0 8 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <h3
          class="pageHeaderTitle"
        >
          cv.codeErrors
        </h3>
        <p
          class="pageHeaderText"
        >
          cv.codeErrorsHeading
        </p>
      </div>
      <div
        class="StyledProps--font StyledProps--main PageSpinner--spinner StyledProps--flex StyledProps--flex-align-center-center"
      >
        <div
          class="Layout--vertical Layout--layout-spacing-medium StyledProps--main PageSpinner--content"
          style="align-items: center; width: 500px;"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
            data-icon="steps-spinner"
          >
            <svg
              height="32"
              viewBox="0 0 128 128"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M122.5 69.25H96.47a33.1 33.1 0 000-10.5h26.05a5.25 5.25 0 010 10.5z"
                  fill="#ff8e02"
                />
                <path
                  d="M112.04 97.83L89.47 84.8a33.1 33.1 0 005.25-9.1l22.57 13.03a5.25 5.25 0 01-5.28 9.1zM88.68 117.35L75.65 94.78a33.1 33.1 0 009.1-5.25l13.02 22.57a5.25 5.25 0 11-9.1 5.25zM58.7 122.57V96.5a33.1 33.1 0 0010.5 0v26.07a5.25 5.25 0 01-10.5 0zM30.1 112.1l13.04-22.57a33.1 33.1 0 009.1 5.25L39.2 117.35a5.25 5.25 0 11-9.1-5.25zM10.6 88.74L33.16 75.7a33.1 33.1 0 005.25 9.1L15.88 97.83a5.25 5.25 0 11-5.25-9.1z"
                  fill="#ffddb3"
                  fill-opacity="0.3"
                />
                <path
                  d="M5.37 58.75h26.06a33.1 33.1 0 000 10.5H5.37a5.25 5.25 0 010-10.5z"
                  fill="#ffd29a"
                  fill-opacity="0.4"
                />
                <path
                  d="M15.85 30.17L38.4 43.2a33.1 33.1 0 00-5.24 9.1L10.6 39.25a5.25 5.25 0 115.25-9.1z"
                  fill="#ffc680"
                  fill-opacity="0.5"
                />
                <path
                  d="M39.2 10.65l13.03 22.57a33.1 33.1 0 00-9.1 5.25l-13-22.57a5.25 5.25 0 119.1-5.25z"
                  fill="#ffbb67"
                  fill-opacity="0.6"
                />
                <path
                  d="M69.2 5.43V31.5a33.1 33.1 0 00-10.5 0V5.42a5.25 5.25 0 1110.5 0z"
                  fill="#ffb04d"
                  fill-opacity="0.7"
                />
                <path
                  d="M97.77 15.9L84.75 38.47a33.1 33.1 0 00-9.1-5.25l13.03-22.57a5.25 5.25 0 119.1 5.25z"
                  fill="#ffa535"
                  fill-opacity="0.8"
                />
                <path
                  d="M117.3 39.26L94.7 52.3a33.1 33.1 0 00-5.25-9.1l22.57-13.03a5.25 5.25 0 015.25 9.1z"
                  fill="#ff991b"
                  fill-opacity="0.9"
                />
                <animatetransform
                  attributeName="transform"
                  calcMode="discrete"
                  dur="1200ms"
                  repeatCount="indefinite"
                  type="rotate"
                  values="0 64 64;30 64 64;60 64 64;90 64 64;120 64 64;150 64 64;180 64 64;210 64 64;240 64 64;270 64 64;300 64 64;330 64 64"
                />
              </g>
            </svg>
          </span>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--font-align-center StyledProps--color StyledProps--color-grey600"
          >
            Loading, please wait...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
